std/node/_stream/async_iterator.ts

```diff
@@ -0,0 +1,265 @@
+import finished from "./end-of-stream.ts";
+import Readable from "./readable.ts";
+
+const kLastResolve = Symbol("lastResolve");
+const kLastReject = Symbol("lastReject");
+const kError = Symbol("error");
+const kEnded = Symbol("ended");
+const kLastPromise = Symbol("lastPromise");
+const kHandlePromise = Symbol("handlePromise");
+const kStream = Symbol("stream");
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+type ReadableIteratorResult = IteratorResult<any>;
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+function isRequest(stream: any) {
+  return stream && stream.setHeader && typeof stream.abort === "function";
+}
+
+//TODO(Soremwar)
+//Should be any implementation of stream
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+function destroyer(stream: any, err?: Error | null) {
+  if (isRequest(stream)) return stream.abort();
+  if (isRequest(stream.req)) return stream.req.abort();
+  if (typeof stream.destroy === "function") return stream.destroy(err);
+  if (typeof stream.close === "function") return stream.close();
+}
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+function createIterResult(value: any, done: boolean): ReadableIteratorResult {
+  return { value, done };
+}
+
+function readAndResolve(iter: ReadableStreamAsyncIterator) {
+  const resolve = iter[kLastResolve];
+  if (resolve !== null) {
+    const data = iter[kStream].read();
+    // We defer if data is null. We can be expecting either 'end' or 'error'.
+    if (data !== null) {
+      iter[kLastPromise] = null;
+      iter[kLastResolve] = null;
+      iter[kLastReject] = null;
+      resolve(createIterResult(data, false));
+    }
+  }
+}
+
+function onReadable(iter: ReadableStreamAsyncIterator) {
+  // We wait for the next tick, because it might
+  // emit an error with `process.nextTick()`.
+  //TODO
+  //This replaces `process.nextTick(readAndResolve, iter);`
```

Comment:
Yes. It is the appropriate API for Deno.

Reply:
I will clean up the comments ASAP